cfme/containers/node.py

```diff
@@ -10,89 +9,68 @@
+from cfme.common import WidgetasticTaggable, TagPageView
+from cfme.containers.provider import (ContainersProvider, Labelable,
+    ContainerObjectAllBaseView, LoggingableView, ContainerObjectDetailsBaseView,
+    click_row)
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator
+from cfme.common.provider_views import ProviderDetailsToolBar
 
-match_page = partial(match_location, controller='container_node', title='Nodes')
 
-# TODO Replace with resource table widget
-resource_locator = "//div[@id='records_div']/table//span[@title='{}']"
+class NodeDetailsToolBar(ProviderDetailsToolBar):
+    web_console = Button('Web Console')
 
 
 class NodeView(ContainerObjectAllBaseView, LoggingableView):
-    TITLE_TEXT = "Nodes"
-
-    nodes = Table(locator="//div[@id='list_grid']//table")
+    SUMMARY_TEXT = "Nodes"
 
     @property
-    def table(self):
-        return self.nodes
+    def nodes(self):
+        return self.table
 
     @property
     def in_cloud_instance(self):
         return (
             self.logged_in_as_current_user and
-            self.navigation.currently_selected == ['Compute', 'Containers', 'Container Nodes'] and
-            match_page()  # No summary, just match controller and title
+            self.navigation.currently_selected == ['Compute', 'Containers', 'Container Nodes']
         )
 
 
 class NodeAllView(NodeView):
     @property
     def is_displayed(self):
-        return (
-            self.in_cloud_instance and
-            match_page(summary='Nodes')
-        )
+        return self.in_cloud_instance and super(NodeAllView, self).is_displayed
 
-    paginator = PaginationPane()
+
+class NodeDetailsView(ContainerObjectDetailsBaseView):
+    toolbar = View.nested(NodeDetailsToolBar)
 
 
 @attr.s
-class Node(Taggable, Labelable, SummaryMixin, BaseEntity):
+class Node(WidgetasticTaggable, Labelable, BaseEntity):
 
     PLURAL = 'Nodes'
+    all_view = NodeAllView
+    details_view = NodeDetailsView
 
     name = attr.ib()
     provider = attr.ib()
 
     @cached_property
     def mgmt(self):
+        """API to use for Nodes"""
         return ApiNode(self.provider.mgmt, self.name)
 
-    def load_details(self, refresh=False):
-        navigate_to(self, 'Details')
-        if refresh:
-            tb.refresh()
-
-    def get_detail(self, *ident):
-        """ Gets details from the details infoblock
-        Args:
-            *ident: Table name and Key name, e.g. "Relationships", "Images"
-        Returns: A string representing the contents of the summary's value.
-        """
-        self.load_details()
-        return InfoBlock.text(*ident)
-
     @classmethod
     def get_random_instances(cls, provider, count=1, appliance=None):
         """Generating random instances."""
         node_list = provider.mgmt.list_node()
         random.shuffle(node_list)
         collection = NodeCollection(appliance)
```

Comment:
this needs some collection v3 conversion. Also, this will case error in appliance is absent.